### PR TITLE
[CFM-787] Fixes errors when leaving the platform

### PR DESCRIPTION
**Pull request by DavidHernandez**
Issue: https://issuetracker.amplexor.com/jira/browse/CFM-787
Previous PR: https://github.com/capacity4dev/capacity4more/pull/1146

### CODE REVIEW

**Comment by ordavidil** [comment]
@DavidHernandez 
Can you add a behat test for that ?

File: project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module [hunk 1]
```diff
@@ -285,7 +285,8 @@ function c4m_user_get_initials($name) {
 function c4m_user_name($user) {
   $wrapper = entity_metadata_wrapper('user', $user);
 
-  if ($user->uid == 0) {
+  // When an user is created, this function is called without uid.
+  if (!isset($user->uid) || $user->uid == 0) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module]: @DavidHernandez 
I'm not sure about this one,
what is the scenario ?

Comment (DavidHernandez) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module]: This test here: https://github.com/capacity4dev/capacity4more/pull/1259/files#diff-f0eba0545c8687cd02f262be9a4bc054R28
Creates a new user to be removed afterwards. Without that change, the code crashes and the user is not created.

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module]: So why not adding first and last name to that user ?

Comment (DavidHernandez) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module]: Just tested it with a first and second name and still fails. That check of the UID happens before checking the first name or the second name.

**Comment by DavidHernandez** [comment]
### This issue now depends on https://github.com/capacity4dev/capacity4more/pull/1280

**Comment by DavidHernandez** [comment]
@ordavidil could you have a look at this issue?

This test here: https://travis-ci.org/capacity4dev/capacity4more/jobs/157613450#L8791 is failing on Travis:
travis
But not on my localhost:
tests-localhost

**Comment by ordavidil** [comment]
@DavidHernandez 
can you add a screenshot of your screen after you press the __Confirm__ button ?

**Comment by DavidHernandez** [comment]
@ordavidil 
out

**Comment by ordavidil** [comment]
@DavidHernandez 
try to replace it to `@javascript` scenario.

**Comment by DavidHernandez** [comment]
@ordavidil seems that the test fails with both @javascript and @api tags. Do you know any other thing we could try?

**Comment by ordavidil** [comment]
let me check it on my machine.

**Comment by ordavidil** [comment]
@DavidHernandez 
I think that travis doesn't know how to handle the batch work that in the background

**Pull request merged**